rfc/20250211-s3-locking-with-conditional-writes.md

```diff
@@ -0,0 +1,136 @@
+# S3 Backend: Locking by using the recently released feature of conditional writes
+
+Issue: https://github.com/opentofu/opentofu/issues/599
+
+Considering the request from the ticket above, and the newly AWS released S3 feature, we can now have state locking without relying on DynamoDB.
+
+The main reasons for such a change could be summarized as follows:
+* Less resources to maintain.
+* Potentially reducing costs by eliminating usage of dynamo db.
+* One less point of failure (removing DynamoDB from the state management)
```

Comment:
Additonal point:
* Easy for other S3 compatible services to implement and enable locking.

Reply:
Added in a3139cf174b05fdcd85dab98b3726874d9a10c4f